src/Libraries/Microsoft.Extensions.AI.Abstractions/AudioTranscription/StreamingAudioTranscriptionUpdateExtensions.cs

```diff
@@ -0,0 +1,167 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+#if NET
+using System.Runtime.InteropServices;
+#endif
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Provides extension methods for working with <see cref="StreamingAudioTranscriptionUpdate"/> instances.
+/// </summary>
+public static class StreamingAudioTranscriptionUpdateExtensions
+{
+    /// <summary>Combines <see cref="StreamingAudioTranscriptionUpdate"/> instances into a single <see cref="AudioTranscriptionCompletion"/>.</summary>
+    /// <param name="updates">The updates to be combined.</param>
+    /// <param name="coalesceContent">
+    /// <see langword="true"/> to attempt to coalesce contiguous <see cref="AIContent"/> items, where applicable,
+    /// into a single <see cref="AIContent"/>, in order to reduce the number of individual content items that are included in
+    /// the manufactured <see cref="AudioTranscriptionChoice"/> instances. When <see langword="false"/>, the original content items are used.
+    /// The default is <see langword="true"/>.
+    /// </param>
+    /// <returns>The combined <see cref="AudioTranscriptionCompletion"/>.</returns>
+    public static AudioTranscriptionCompletion ToAudioTranscriptionCompletion(
+        this IEnumerable<StreamingAudioTranscriptionUpdate> updates, bool coalesceContent = true)
+    {
+        _ = Throw.IfNull(updates);
+
+        AudioTranscriptionCompletion completion = new([]);
```

Comment:
AudioTranscriptionCompletion => AudioTranscription ?

Reply:
`AudioTranscriptionResponse`, following the `Response/Request`. Made sense keeping this for the other modalities also, no?